tests/charms/tls_certificates_interface/v1/test_tls_certificates_v1.py

```diff
@@ -86,6 +91,47 @@ def _load_relation_data(raw_relation_data: dict) -> dict:
     return certificate_data
 
 
+def validate_induced_data_from_pfx_is_equal_to_initial_data(
+    pfx_file: bytes,
+    password: str,
+    initial_certificate: bytes,
+    initial_private_key: bytes,
+):
+    (
+        induced_private_key_object,
+        induced_certificate_object,
+        additional_certificate,
+    ) = pkcs12.load_key_and_certificates(pfx_file, password.encode())
+
+    initial_private_key_object = load_pem_private_key(
+        initial_private_key,
+        password=None,
+    )
+    initial_public_key_object = initial_private_key_object.public_key()
+    induced_public_key_object = induced_private_key_object.public_key()  # type: ignore[union-attr]
+    induced_certificate = induced_certificate_object.public_bytes(  # type: ignore[union-attr]
+        encoding=serialization.Encoding.PEM
+    )
+
+    initial_public_key = initial_public_key_object.public_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PublicFormat.PKCS1,
+    )
+    induced_public_key = induced_public_key_object.public_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PublicFormat.PKCS1,
+    )
+    induced_private_key = induced_private_key_object.private_bytes(  # type: ignore[union-attr]
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    )
```

Comment:
Some nitpicking, I think reorganizing the lines like this makes it a bit easier to follow. Feel free to ignore.
```suggestion
    initial_private_key_object = load_pem_private_key(
        initial_private_key,
        password=None,
    )
    induced_private_key = induced_private_key_object.private_bytes(  # type: ignore[union-attr]
        encoding=serialization.Encoding.PEM,
        format=serialization.PrivateFormat.TraditionalOpenSSL,
        encryption_algorithm=serialization.NoEncryption(),
    )
    
    initial_public_key_object = initial_private_key_object.public_key()
    initial_public_key = initial_public_key_object.public_bytes(
        encoding=serialization.Encoding.PEM,
        format=serialization.PublicFormat.PKCS1,
    )
    
    induced_public_key_object = induced_private_key_object.public_key()  # type: ignore[union-attr]
    induced_public_key = induced_public_key_object.public_bytes(
        encoding=serialization.Encoding.PEM,
        format=serialization.PublicFormat.PKCS1,
    )
    
    induced_certificate = induced_certificate_object.public_bytes(  # type: ignore[union-attr]
        encoding=serialization.Encoding.PEM
    )
```